app/controllers/works_controller.rb

```diff
@@ -0,0 +1,67 @@
+class WorksController < ApplicationController
+    def index
+        @works = Work.all
+
+        @albums = @works.where(category: 'album')
+        @books = @works.where(category: 'book')
+        @movies = @works.where(category: 'movie')
+    end
+
+    def show
+        @work = Work.find_by(id: params[:id])
+
+        if @work.nil?
+            head :not_found
+            return
+        end 
+    end
+
+    def new
+        @work = Work.new
+    end
+
+    def edit
+        @work = Work.find_by(id: params[:id])
+
+        if @work.nil?
+            redirect_to works_path
+            return
+        end
```

Comment:
Consider adding this code to a controller filter like `find_work` to DRY up your code.